src/abstractdatatable/join.jl

```diff
@@ -15,76 +15,6 @@ similar_nullable{T,R}(dv::CategoricalArray{T,R}, dims::@compat(Union{Int, Tuple{
 similar_nullable(dt::AbstractDataTable, dims::Int) =
     DataTable(Any[similar_nullable(x, dims) for x in columns(dt)], copy(index(dt)))
 
-function join_idx(left, right, max_groups)
-    ## adapted from Wes McKinney's full_outer_join in pandas (file: src/join.pyx).
-
-    # NULL group in location 0
-
-    left_sorter, where, left_count = groupsort_indexer(left, max_groups)
-    right_sorter, where, right_count = groupsort_indexer(right, max_groups)
-
-    # First pass, determine size of result set
-    tcount = 0
-    rcount = 0
-    lcount = 0
-    for i in 1:(max_groups + 1)
-        lc = left_count[i]
-        rc = right_count[i]
-
-        if rc > 0 && lc > 0
-            tcount += lc * rc
-        elseif rc > 0
-            rcount += rc
-        else
-            lcount += lc
-        end
-    end
-
-    # group 0 is the NULL group
-    tposition = 0
-    lposition = 0
-    rposition = 0
-
-    left_pos = 0
-    right_pos = 0
-
-    left_indexer = Vector{Int}(tcount)
-    right_indexer = Vector{Int}(tcount)
-    leftonly_indexer = Vector{Int}(lcount)
-    rightonly_indexer = Vector{Int}(rcount)
-    for i in 1:(max_groups + 1)
-        lc = left_count[i]
-        rc = right_count[i]
-        if rc == 0
-            for j in 1:lc
-                leftonly_indexer[lposition + j] = left_pos + j
-            end
-            lposition += lc
-        elseif lc == 0
-            for j in 1:rc
-                rightonly_indexer[rposition + j] = right_pos + j
-            end
-            rposition += rc
-        else
-            for j in 1:lc
-                offset = tposition + (j-1) * rc
-                for k in 1:rc
-                    left_indexer[offset + k] = left_pos + j
-                    right_indexer[offset + k] = right_pos + k
-                end
-            end
-            tposition += lc * rc
-        end
-        left_pos += lc
-        right_pos += rc
-    end
-
-    ## (left_sorter, left_indexer, leftonly_indexer,
-    ##  right_sorter, right_indexer, rightonly_indexer)
-    (left_sorter[left_indexer], left_sorter[leftonly_indexer],
-     right_sorter[right_indexer], right_sorter[rightonly_indexer])
-end
-
 function sharepools{S,N}(v1::Union{CategoricalArray{S,N}, NullableCategoricalArray{S,N}},
```

Comment:
Coveralls indicates this function isn't tested anymore, and I think it isn't used at all now. Better remove it. `==` for `DataTableRow` and `permute`/`ipermute` aren't covered either: we need more tests for them. Have a look at the coverage report to spot other smaller methods which need testing.